exporter/collector/googlemanagedprometheus/factory.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googlemanagedprometheus
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+
+	"github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/collector"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr        = "googlemanagedprometheus"
+	defaultTimeout = 12 * time.Second // Consistent with Cloud Monitoring's timeout
+)
+
+// NewFactory creates a factory for the googlemanagedprometheus exporter
+func NewFactory() component.ExporterFactory {
+	return component.NewExporterFactory(
+		typeStr,
+		createDefaultConfig,
+		component.WithMetricsExporter(createMetricsExporter),
+	)
+}
+
+// createDefaultConfig creates the default configuration for exporter.
+func createDefaultConfig() config.Exporter {
+	return &Config{
+		ExporterSettings: config.NewExporterSettings(config.NewComponentID(typeStr)),
+		TimeoutSettings:  exporterhelper.TimeoutSettings{Timeout: defaultTimeout},
+		RetrySettings:    exporterhelper.NewDefaultRetrySettings(),
+		QueueSettings:    exporterhelper.NewDefaultQueueSettings(),
+		GMPConfig: GMPConfig{
+			UserAgent: "opentelemetry-collector-contrib {{version}}",
```

Comment:
+1 to a consistent layout, however also +1 to making sure we know "opentelemetry" is involved somewhere in the UA (for diagnosing bugs).

Reply:
changed to opentelemetry-collector-contrib/{version} to match formatting.

Reply:
Do you also want to structure it so it can be extended to include the source in the future? e.g., it may be useful to know whether the metrics are coming from Ops Agent vs otherwise.

Reply:
It is settable via config options.  The ops-agent could set it to: `user_agent: ops-agent-gmp/{{version}}` if they wanted, for example.